### DISCUSSION TITLE
OPA Test Error: detected overlapping roots in bundle manifest

### DISCUSSION THREAD

**Discussion by mholliday1** [open-policy-agent/community#267] [OPA and Rego] [answered]
Hello!👋

I am trying to run `opa test` on two specific directories bundled together. However I am running into the error: `detected overlapping roots in bundle manifest with: [./global ./policy]` but they don't seem to be overlapping in my mind. So I'm not sure how to fix it.

The command I am using is:
`opa test --bundle ./global ./policy --verbose`

I've read through this issue: [#4428](https://github.com/open-policy-agent/opa/issues/4428) and this feature request: [#4724](https://github.com/open-policy-agent/opa/issues/4724) but didn't really find/understand a solution.

I'd appreciate if someone could point me in the right direction!

**Comment by ashutosh-narkar** [comment, marked as answer]
What version of OPA are you using? Also can you help us repro the issue?

**Reply by mholliday1** [reply]
I am using version 0.41.0

I attached a simplified version of the structure which looks essentially like this:

```
root
    |__ global
    |      |__ folder
    |            |__ unit_test_data_folder
    |            |__ test_file
    |            |__ policy_file
    |__ policy
           |__ folder
                  |__ unit_test_data_folder
                  |__ test_file
                  |__ policy_file
```

I'm using the same policy and tests twice but in two different folders just to show as an example of the situation but all that matters is I want to run the tests in the `policy` and `global` folders. In this situation I could run the command `opa test --bundle . --verbose` and it works perfectly, however in practice I actually have another folder at the same level of the `policy` and `global` folders that I do not want `opa test` to run in because when it does it causes other errors.

The first thing I tried was using the `ignore` flag but it didn't work and I also read another issue that said the flag doesn't work with bundles.

Thanks for taking a look at this!
[test.zip](https://github.com/open-policy-agent/community/files/9530964/test.zip)

**Reply by anderseknert** [reply]
Great timing.. the `--ignore` flag was fixed for bundles in OPA [v0.44.0](https://github.com/open-policy-agent/opa/releases/tag/v0.44.0), which was released 22 hours ago :) 

If you want to fix the problem "correctly" — the issue is that bundles by default think of themselves as owners of _all_ roots. So both bundles will think they own both of the roots. You can avoid this by placing a `.manifest` file in each directory, where they'd say something like:
```json
{
  "roots": ["global"]
}
``` 
and
```json
{
  "roots": ["policy"]
}
``` 
respectively... see the docs on the bundle file format for more details.

**Reply by mholliday1** [reply]
Oh awesome, I will take a closer look at those docs, thank you so much!

**Comment by anderseknert** [comment]
Hmm, would you not need to repeat the `--bundle` flag for each directory? Just thinking out loud here 😄

**Reply by mholliday1** [reply]
Like running `opa test --bundle ./global --bundle ./policy --verbose` instead? Sadly got the same error but thank you for the suggestion!
